common/cards/hermits/pharaoh-rare.ts

```diff
@@ -0,0 +1,109 @@
+import {HERMIT_CARDS} from '..'
+import {CardPosModel} from '../../models/card-pos-model'
+import {GameModel} from '../../models/game-model'
+import {getNonEmptyRows} from '../../utils/board'
+import {flipCoin} from '../../utils/coinFlips'
+import HermitCard from '../base/hermit-card'
+
+class PharaohRareHermitCard extends HermitCard {
+	constructor() {
+		super({
+			id: 'pharaoh_rare',
+			numericId: 173,
+			name: 'Pharaoh',
+			rarity: 'rare',
+			hermitType: 'balanced',
+			health: 300,
+			primary: {
+				name: 'Targét',
+				cost: ['balanced'],
+				damage: 50,
+				power: null,
+			},
+			secondary: {
+				name: 'Xibalba',
+				cost: ['balanced', 'balanced'],
+				damage: 80,
+				power:
+					"Flip a coin. If heads, can give up to 80hp to AFK Hermit. Health given is equal to damage during attack. Can't heal other Pharaohs.",
+			},
+		})
+	}
+
+	override onAttach(game: GameModel, instance: string, pos: CardPosModel) {
+		const {player} = pos
+		const instanceKey = this.getInstanceKey(instance)
+
+		player.hooks.afterAttack.add(instance, (attack) => {
```

Comment:
Might be wrong, but is this called every attack or only start all attacks have run